osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -36,92 +38,101 @@ public class GameplayClockContainer : Container, IAdjustableClock, IGameplayCloc
+            }
+        }
 
-        public virtual IEnumerable<double> NonGameplayAdjustments => Enumerable.Empty<double>();
+        private double startTime;
 
-        /// <summary>
-        /// The final clock which is exposed to gameplay components.
-        /// </summary>
-        protected IFrameBasedClock FramedClock { get; private set; }
+        public virtual IEnumerable<double> NonGameplayAdjustments => Enumerable.Empty<double>();
 
         private readonly BindableBool isPaused = new BindableBool(true);
 
         /// <summary>
         /// The adjustable source clock used for gameplay. Should be used for seeks and clock control.
+        /// This is the final clock exposed to gameplay components as an <see cref="IGameplayClock"/>.
         /// </summary>
-        private readonly DecoupleableInterpolatingFramedClock decoupledClock;
+        protected readonly FramedBeatmapClock GameplayClock;
+
+        protected override Container<Drawable> Content { get; } = new Container { RelativeSizeAxes = Axes.Both };
 
         /// <summary>
         /// Creates a new <see cref="GameplayClockContainer"/>.
         /// </summary>
         /// <param name="sourceClock">The source <see cref="IClock"/> used for timing.</param>
-        public GameplayClockContainer(IClock sourceClock)
+        /// <param name="applyOffsets">Whether to apply platform, user and beatmap offsets to the mix.</param>
+        public GameplayClockContainer(IClock sourceClock, bool applyOffsets = false)
         {
             SourceClock = sourceClock;
 
             RelativeSizeAxes = Axes.Both;
 
-            decoupledClock = new DecoupleableInterpolatingFramedClock { IsCoupled = false };
-            IsPaused.BindValueChanged(OnIsPausedChanged);
-
-            // this will be replaced during load, but non-null for tests which don't add this component to the hierarchy.
-            FramedClock = new FramedClock();
-        }
-
-        protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnlyDependencyContainer parent)
-        {
-            var dependencies = new DependencyContainer(base.CreateChildDependencies(parent));
-
-            FramedClock = CreateGameplayClock(decoupledClock);
-
-            dependencies.CacheAs<IGameplayClock>(this);
+            InternalChildren = new Drawable[]
+            {
+                GameplayClock = new FramedBeatmapClock(sourceClock, applyOffsets) { IsCoupled = false },
+                Content
+            };
 
-            return dependencies;
+            IsPaused.BindValueChanged(OnIsPausedChanged);
         }
 
         /// <summary>
-        /// Starts gameplay.
+        /// Starts gameplay and marks un-paused state.
         /// </summary>
         public virtual void Start()
         {
             ensureSourceClockSet();
 
-            if (!decoupledClock.IsRunning)
+            isPaused.Value = false;
+
+            // the clock may be stopped via internal means (ie. not via `IsPaused`).
```

Comment:
The "internal means" here being...? It is mentioned what the means are *not*, but that kind of doesn't help to know what to watch out for.

Reply:
One case would be spectator clock, when catching up for frames. Basically we have this weird situation where `IsPaused` is tracking the "user" paused state, which doesn't necessarily match `IsRunning`. I agree it's bad. I'll expand on the comment for now.

Reply:
I've updated the inline comment to include a reference to the local case where this occur.

Reply:
This is gone in the latest commit.